src/Type/MixedType.php

```diff
@@ -484,6 +486,15 @@ public function toAbsoluteNumber(): Type
 
 	public function toInteger(): Type
 	{
+		if ($this->subtractedType !== null) {
+			if ($this->subtractedType->isSuperTypeOf(new ConstantIntegerType(0))->yes()) {
```

Comment:
I went with this special case for `0`, because a non-zero int can later on easily be turned in more precise types (like a non-falsey-string etc.)

additionally I think its pretty common to subtract 0 from a given int.